File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/UserSkillsScoreForm.ts
import { LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface UserSkillsScoreForm {
    SkillCategoryId: LookupEditor;
    SkillId: LookupEditor;
}

export class UserSkillsScoreForm extends PrefixedContext {
    static readonly formKey = 'Masters.UserSkillsScore';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UserSkillsScoreForm.init)  {
            UserSkillsScoreForm.init = true;

            var w0 = LookupEditor;

            initFormType(UserSkillsScoreForm, [
                'SkillCategoryId', w0,
                'SkillId', w0
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Masters/UserSkillsScore/UserSkillsScoreForm.cs
using Serenity.ComponentModel;

namespace GXpert.Masters.Forms;

[FormScript("Masters.UserSkillsScore")]
[BasedOnRow(typeof(UserSkillsScoreRow), CheckNames = true)]
public class UserSkillsScoreForm
{
    public int SkillCategoryId { get; set; }
    public int SkillId { get; set; }
    public float Score { get; set; }   
}